tensorbay/cli/utility.py

```diff
@@ -26,95 +26,213 @@
+        if config_parser.has_section("config"):
+            config_section = config_parser["config"]
+            if "timeout" in config_section:
+                client_config.timeout = config_section.getint("timeout")
+            if "is_internal" in config_section:
+                client_config.is_internal = config_section.getboolean("is_internal")
+            if "max_retries" in config_section:
+                client_config.max_retries = config_section.getint("max_retries")
+
+    def generate_profiles(self) -> Iterable[Tuple[str, str, str]]:
+        """Get all profiles and corresponding profile information.
+
+        Yields:
+            A tuple containing profile name and the corresponding accesskey and url read from
+            the config file.
+
+        """
+        try:
+            profiles = self.config_parser["profiles"]
+        except KeyError:
+            return
+        yield from map(
+            lambda profile_info: (profile_info[0], *self._parse_profile_info(profile_info[1])),
+            profiles.items(),
+        )
 
-    Returns:
-        A tuple containing the accessKey and the url of profile_name read from the config file.
+    def get_gas(self, access_key: Optional[str] = None, url: Optional[str] = None) -> GAS:
+        """Load an object of :class:`~tensorbay.client.gas.GAS`.
 
-    """
-    if not config_parser.has_option("profiles", profile_name):
-        error(
-            f"{profile_name} does not exist"
-            f"\n\nPlease use 'gas -p {profile_name} auth [accessKey]' to create profile"
-        )
-    values = config_parser["profiles"][profile_name].split("\n")
-    if len(values) == 2:
-        return values[1], ""
-    return values[1], values[2]
+        We will read accessKey and URL from the appointed profile_name and login gas.
 
+        Arguments:
+            access_key: The accessKey of gas.
+            url: The login URL.
 
-def get_gas(access_key: str, url: str, profile_name: str, config_parser: ConfigParser) -> GAS:
-    """Load an object of :class:`~tensorbay.client.gas.GAS`.
+        Returns:
+            Gas client logged in with accessKey and URL.
 
-    We will read accessKey and URL from the appointed profile_name and login gas.
+        """
+        self._set_request_config()
 
-    Arguments:
-        access_key: The accessKey of gas.
-        url: The login URL.
-        profile_name: The environment to login.
-        config_parser: The config parser read from the context object.
+        if access_key is None:
+            access_key, url = self.access_key, self.url
 
-    Returns:
-        Gas client logged in with accessKey and URL.
+        if not access_key and not url:
+            access_key, url = self.read_profile()
 
-    """
-    _set_request_config(config_parser)
-    if not access_key and not url:
-        access_key, url = read_profile(profile_name, config_parser)
+        if not access_key:
+            error("AccessKey should be appointed")
```

Comment:
should we add more detailed guide about how to appoint accesskey in the error message?